drivers/watchdog/wdt_nrfx.c

```diff
@@ -166,6 +166,35 @@ static const struct wdt_driver_api wdt_nrfx_driver_api = {
 	.feed = wdt_nrf_feed,
 };
 
+#define WDT(idx) DT_NODELABEL(wdt##idx)
+
+#if defined(CONFIG_WDT_NRFX_NO_IRQ)
+#define WDT_NRFX_WDT_DEVICE(idx)					       \
```

Comment:
I guess instead of ifdefing whole `WDT_NRFX_WDT_DEVICE` we can utilize `COND_CODE_1(IS_ENABLED(CONFIG_WDT_NRFX_NO_IRQ))` and have `WDT_NRFX_WDT_DEVICE` defined only once. See: https://github.com/zephyrproject-rtos/zephyr/blob/f2259ccd7ab9b772ad1936c999a363323144b377/drivers/spi/spi_nrfx_spim.c#L714 for reference

Reply:
Changed